omegaconf/base.py

```diff
@@ -399,7 +400,7 @@ def _select_impl(
             if ret is not None and not isinstance(ret, Container):
                 parent_key = ".".join(split[0 : i + 1])
                 child_key = split[i + 1]
-                raise ConfigKeyError(
+                raise ConfigTypeError(
```

Comment:
Not going to push for it, but just a suggestion to restrict this new feature to `AnyNodes` that are `None` (because it still doesn't make much sense to me to let people try to select into other nodes):
* Check here if `isinstance(ret, AnyNode) and ret._is_none()` and if it is, return something that will make `select()` return `None` (or the default value). It may be worth looking into #674 at the same time since it will probably require some changes regarding when to return the default value, which might also mean changing what to return here.
* Otherwise, raise the exception (and don't catch it in the caller)

Reply:
The contact of OmegaConf.select is that it returns None if the selection cannot be performed.
It makes it harder to use if it sometimes raises an exception due to subtle reasons.

Hydra is using OmegaConf.select to determine if a config node exists in a path. raising an exception there makes the API significantly more cumbersome to use.
We can potentially introduce a specific API to do that in the future (but certainly not for 2.1), but in general I think using select for it is very reasonable.

Reply:
Got it